src/core/str/py_str.cc

```diff
@@ -140,9 +63,32 @@ static oobj split_into_nhot(const PKArgs& args) {
   return Frame::oframe(res);
 }
 
+DECLARE_PYFN(&split_into_nhot)
+    ->name("split_into_nhot")
+    ->docs(dt::doc_str_split_into_nhot)
+    ->n_positional_args(1)
+    ->n_keyword_args(2)
+    ->arg_names({"frame", "sep", "sort"});
 
-void DatatableModule::init_methods_str() {
-  ADD_FN(&split_into_nhot, args_split_into_nhot);
+
+
+static oobj split_into_nhot_deprecated(const XArgs& args) {
+  auto w = DeprecationWarning();
+  w << "Function `dt.split_into_nhot()` is deprecated since version 1.0.0, "
+       "and will be removed in version 1.1.0.\n"
+       "Please use `dt.str.split_into_nhot()` instead.";
+  w.emit_warning();
+  return split_into_nhot(args);
 }
 
+DECLARE_PYFN(&split_into_nhot_deprecated)
+    ->name("split_into_nhot_depr")
```

Comment:
Didn't notice you left the old name here. May be let's use `split_into_nhot_depr` or `split_into_nhot_deprecated` everywhere?